registry/jupyter/datascience-notebook/container.yaml

```diff
@@ -3,10 +3,15 @@ url: https://hub.docker.com/r/jupyter/datascience-notebook
 maintainer: '@vsoch'
 description: Jupyter Datascience Notebook from https://github.com/jupyter/docker-stacks
 latest:
-  "4.0": sha256:5e5bf78bfa351c0255cea5269a0461afbf6b50b51e923a7436229208ea8487f9
+  b78201340369: sha256:027e159b6780294aa930623f0f7e6c5094972be48d5e15027024a811fdc7e5c7
```

Comment:
@muffato this looks like a red flag - I'm wondering if we should try to filter out patterns that look like commits.

Reply:
Yes, you're right. This looks like a commit or a checksum. It's not something that `shpc` could rank against the other tags
(especially this particular one is 2 year old https://hub.docker.com/layers/datascience-notebook/jupyter/datascience-notebook/b78201340369/images/sha256-027e159b6780294aa930623f0f7e6c5094972be48d5e15027024a811fdc7e5c7?context=explore whereas other tags are less then a week old. Hopefully they're legacy tags from a time when they didn't have proper versioning in place ?)

Reply:
And to answer https://github.com/singularityhub/singularity-hpc/pull/541#issuecomment-1090781733 

Let's maybe allow strings of length <=8 made of digits only because some project may use timestamps e.g. 20220406.

Option 1. To exclude commit IDs: ignore the strings of length > 8 that are only composed of digits + hex characters only.

Option 2. Am I understanding right that you need some integers to be able to rank the versions ? You could do something after the parsing and splitting. Could we say that the tuple must start with at least 1 integer and that the lengths of the integers must be 8 characters at most ? So it would:
- allow '4.b78201340369' because it looks like a version 4 something, which can still be compared against other version numbers
- exclude all commit IDs that include a letter
- include strings that are entirely made of digits as long as they're not too long